Add Example for Using Rendering Module in Documentation (#100)
PsiQ/qref: docs/library/userguide.md

If you are familiar with [Pydantic](https://docs.pydantic.dev/latest/), you might find
it easier to work with QREF Pydantic models instead of interacting with JSON schema directly.

> In the example below, we create an instance of [`SchemaV1`](qref.experimental.SchemaV1) model from

Reviewer: [issue] This link doesn't work, please change back to what it was.

The `qref-render` tool supports `yaml` and `json` input formats, and all
output formats supported by [graphviz](https://graphviz.org/).

> If you prefer to use QREF's rendering capabilities from a Python script instead of the CLI, you can use the [`qref.experimental.rendering`](qref.experimental.rendering) module,  which performs the same task as `qref-render`. Here, we demonstrate how to use the rendering module to visualize quantum circuits for preparing arbitrary quantum states in alias sampling. To learn more about the algorithm, please refer to the tutorial for [Bartiq](https://psiq.github.io/bartiq/latest/tutorials/02_alias_sampling_basic/).

Reviewer: ```suggestion
If you prefer to use QREF's rendering capabilities from a Python script instead of the CLI, you can use the [`qref.experimental.rendering`](qref.experimental.rendering) module,  which performs the same task as `qref-render`. Here, we demonstrate how to use the rendering module to visualize quantum circuits for preparing arbitrary quantum states in alias sampling. To learn more about the algorithm, please refer to the tutorial for [Bartiq](https://psiq.github.io/bartiq/latest/tutorials/02_alias_sampling_basic/) – our library for symbolic resource estimation.
```

> We will use the `yaml` file `alias_sampling_detailed_fixed_resources.yaml` as input to generate a graph of the circuit:

Reviewer: ```suggestion
We will use the `yaml` file `alias_sampling_detailed_fixed_resources.yaml` as input to generate a graph representing this algorithm:
```

```python
import yaml
from qref import SchemaV1
from qref.experimental.rendering import to_graphviz

# Load the YAML file
with open("../examples/alias_sampling_detailed_fixed_resources.yaml", "r") as f:
    data = yaml.safe_load(f)

# Validate the schema and convert to Graphviz object
program = SchemaV1.model_validate(data)
gv_object = to_graphviz(program)

# Render the Graphviz object to a PNG file

> gv_object.render("as",format="png")

Reviewer: [nitpick] I'd change the name from `as` to `alias_sampling` – it might not be immediately obvious to someone what `as` stands for.
